packages/webpack-rsc/src/webpack-rsc-server-loader.cts

```diff
@@ -137,16 +137,27 @@ function isDirective(
     t.isDirectiveLiteral(directive.value) && directive.value.value === value;
 }
 
-function getExportName(node: t.Node): string | undefined {
+function getFunctionExportName(node: t.Node): string | undefined {
   if (t.isExportNamedDeclaration(node)) {
     if (t.isFunctionDeclaration(node.declaration)) {
       return node.declaration.id?.name;
     }
 
     if (t.isVariableDeclaration(node.declaration)) {
-      const id = node.declaration.declarations[0]?.id;
+      const [variableDeclarator] = node.declaration.declarations;
 
-      return t.isIdentifier(id) ? id.name : undefined;
+      if (variableDeclarator) {
+        const {id, init} = variableDeclarator;
+
+        if (
+          t.isIdentifier(id) &&
+          (t.isArrowFunctionExpression(init) || t.isFunctionExpression(init))
```

Comment:
will this catch
```
async function foo() { ... };
export const bar = foo;
```
?
FWIW i think next just does
```js
if (typeof NAME === 'function') {
  registerServerReference(NAME, 'NAME' ...);
}
```
because it's hard to analyze this statically

Reply:
No, this does not cover all possible scenarios. It's pretty opinionated regarding what code style it supports, e.g. I also don't want to support default exports.

Reply:
...or inline 'use server' directives

Reply:
```js
if (typeof NAME === 'function') {
  registerServerReference(NAME, 'NAME' ...);
} else {
  errorBadExport('NAME');
}
```
Hmm, I am considering this now... 
Thanks for bringing this to my attention.

Reply:
But why would you error in the else case?

Reply:
i think it's because the expectation is that a "use server" module exports server references, and importing other things from it might act weird, because they won't get tranformed. like `export const foo = 1` would work, but arbitrary non-function exports might not -- what if someone does `export const dbClient = myDb.connect()`? so easier to just blanket-ban than let the user run into puzzling errors i think

Reply:
Yeah, makes sense.